cmd/kubeadm/app/util/version.go

```diff
@@ -83,3 +91,33 @@ func KubernetesVersionToImageTag(version string) string {
 	allowed := regexp.MustCompile(`[^-a-zA-Z0-9_\.]`)
 	return allowed.ReplaceAllString(version, "_")
 }
+
+// KubernetesValidateVersion is a helper function that can validate
+// if user requested release version or version from CI build area
+// and then return validated and/or resolved from keywords like "latest".
+//
+// Examples of supported versions:
+// v1.7.1 -- released version.
+// release/v1.7.1 -- same as above.
+// stable-1.7 or release/stable-1.7 -- labeled official releases
+// ci/v1.8.0-alpha.1.123+012345678 -- CI explicit build version
+// ci/latest-1.8 or ci-cross/latest-1.8 -- CI labeled builds
+//
+// Returns: version, build is release (false) or CI (true), error.
+func KubernetesValidateVersion(version string) (string, bool, error) {
```

Comment:
I find the relationship between KubernetesValidateVersion and KubernetesReleaseVersion kind of strange.
What I think we want is this:
 - Have a function that takes `cfg.KubernetesVersion` (may be a label) and just return the resulting "real" version we can use. Basically what `KubernetesReleaseVersion` is today.

This would do the following:
If it already is a valid version, return
If it matches `(release/)?(stable|latest)(-1.[0-9]{1})?`, unconditionally trim the `release/` prefix and fetch real version from `dl.k8s.io/release/%s.txt` and return
If it matches `ci/latest(-1.[0-9]{1})?`, unconditionally trim the `ci/` prefix and fetch the real version from `dl.k8s.io/ci-cross/%s.txt` and return

That will do what we want. Feel free to create private helper functions for that main function.